src/utils/utils_common.h

```diff
@@ -19,11 +19,45 @@
 extern "C" {
 #endif
 
-#ifdef _WIN32
+#ifdef _WIN32 /* Windows */
+
 #define __TLS __declspec(thread)
-#else
+
+static inline char *os_getenv(const char *name) {
+    char *buffer;
+    size_t numberOfElements;
+    errno_t err = _dupenv_s(&buffer, &numberOfElements, name);
+    if (err) {
+        return NULL;
+    }
+
+    return buffer;
+}
+
+static inline void os_free_getenv(char *val) { free(val); }
+
+#else /* Linux */
+
 #define __TLS __thread
-#endif
+
+static inline char *os_getenv(const char *name) { return getenv(name); }
+static inline void os_free_getenv(const char *val) {
+    (void)val; // unused
+}
+
+#endif /* _WIN32 */
+
+// check if we are running in the proxy library
+static inline int is_running_in_proxy_lib(void) {
```

Comment:
Btw, it would be nice to have a test for such scenario.

Reply:
> Will this work if someone links with proxy_lib? (I think we should support this scenario - from what I understand, this is how it will work on windows) Or is there no problem in that case?

Do you mean the dynamic linking instead of LD_PRELOD?

Reply:
Yes